external/storm-eventhubs/src/main/java/org/apache/storm/eventhubs/spout/StringEventDataScheme.java

```diff
@@ -44,21 +40,23 @@
 public class StringEventDataScheme implements IEventDataScheme {
 
   private static final long serialVersionUID = 1L;
+  private static final Logger logger = LoggerFactory.getLogger(StringEventDataScheme.class);
 
   @Override
-  public List<Object> deserialize(Message message) {
+  public List<Object> deserialize(EventData eventData) {
     final List<Object> fieldContents = new ArrayList<Object>();
-
-    for (Section section : message.getPayload()) {
-      if (section instanceof Data) {
-        Data data = (Data) section;
-        fieldContents.add(new String(data.getValue().getArray()));
-      } else if (section instanceof AmqpValue) {
-        AmqpValue amqpValue = (AmqpValue) section;
-        fieldContents.add(amqpValue.getValue().toString());
+    String messageData = "";
+    if(eventData.getBytes()!=null)
+      messageData = new String (eventData.getBytes());
+    else if(eventData.getObject()!=null){
+      try{
+        messageData = new String(Serializedeserializeutil.serialize(eventData.getObject()),Charset.defaultCharset());
+      }catch (IOException e){
+        logger.error("Failed to serialize object"+e.toString());
```

Comment:
>object [](start = 42, length = 6)

nit: i would pitch the msg a bit more specific, like - failed to Serialize EventData payload